### PR TITLE
NOISSUE - Make MQTT Broker Configurable

### PR DESCRIPTION
**Pull request by rodneyosodo**
Signed-off-by: Rodney Osodo <socials@rodneyosodo.com>

### What does this do?
This PR makes the MQTT broker configurable either as VerneMQ or NATS. NATS is a lightweight and highly performant message broker that offers MQTT support, making it a suitable replacement for VerneMQ in this context.

### Which issue(s) does this PR fix/relate to?
No issue

### List any changes that modify/break current functionality
1. Add docker profiles to switch between different modes. The available profiles are:
	- `vernemq_nats`: Uses `vernemq` as MQTT_BROKER and `nats` as MESSAGE_BROKER.
	- `vernemq_rabbitmq`: Uses `vernemq` as MQTT_BROKER and `rabbitmq` as MESSAGE_BROKER.
	- `nats_nats`: Uses `nats` as both MQTT_BROKER and MESSAGE_BROKER.
	- `nats_rabbitmq`: Uses `nats` as MQTT_BROKER and `rabbitmq` as MESSAGE_BROKER.

3. Integration of NATS as the MQTT broker.

### Have you included tests for your changes?
Tested manually

### Did you document any new/modified functionality?
https://github.com/mainflux/docs/pull/159

### Notes
To be merged after https://github.com/mainflux/mainflux/pull/1903 and https://github.com/mainflux/mainflux/pull/1902

- NATS follows MQTT v3.1.1 specification
- Limitations regarding Nats MQTT support can be found [here](https://docs.nats.io/running-a-nats-service/configuration/mqtt#limitations) 

#### How to test

1. NATS as MQTT Broker and NATS as Message Broker

	```bash
	MF_MESSAGE_BROKER_TYPE=nats MF_MQTT_BROKER_TYPE=nats make run up args="-d"
	MF_MESSAGE_BROKER_TYPE=nats MF_MQTT_BROKER_TYPE=nats make run_addons up args="-d"
	```

2. VerneMQ as MQTT Broker and NATS as Message Broker

	```bash
	MF_MESSAGE_BROKER_TYPE=nats MF_MQTT_BROKER_TYPE=vernemq make run up args="-d"
	MF_MESSAGE_BROKER_TYPE=nats MF_MQTT_BROKER_TYPE=vernemq make run_addons up args="-d"
	```

3. NATS as MQTT Broker and RabbitMQ as Message Broker

	```bash
	MF_MESSAGE_BROKER_TYPE=rabbitmq MF_MQTT_BROKER_TYPE=nats make run up args="-d"
	MF_MESSAGE_BROKER_TYPE=rabbitmq MF_MQTT_BROKER_TYPE=nats make run_addons up args="-d"
	```

4. VerneMQ as MQTT Broker and RabbitMQ as Message Broker

	```bash
	MF_MESSAGE_BROKER_TYPE=rabbitmq MF_MQTT_BROKER_TYPE=vernemq make run up args="-d"
	MF_MESSAGE_BROKER_TYPE=rabbitmq MF_MQTT_BROKER_TYPE=vernemq make run_addons up args="-d"
	```

### CODE REVIEW

File: pkg/messaging/rabbitmq/pubsub_test.go [hunk 1]
```diff
@@ -419,8 +419,6 @@ func TestPubSub(t *testing.T) {
 			assert.Equal(t, expectedMsg.Channel, receivedMsg.Channel, fmt.Sprintf("%s: expected %+v got %+v\n", tc.desc, &expectedMsg, receivedMsg))
 			assert.Equal(t, expectedMsg.Created, receivedMsg.Created, fmt.Sprintf("%s: expected %+v got %+v\n", tc.desc, &expectedMsg, receivedMsg))
 			assert.Equal(t, expectedMsg.Protocol, receivedMsg.Protocol, fmt.Sprintf("%s: expected %+v got %+v\n", tc.desc, &expectedMsg, receivedMsg))
-			assert.Equal(t, expectedMsg.Publisher, receivedMsg.Publisher, fmt.Sprintf("%s: expected %+v got %+v\n", tc.desc, &expectedMsg, receivedMsg))
```
Comment (SammyOina) [inline pkg/messaging/rabbitmq/pubsub_test.go:422]: why are these removed

File: mqtt/handler.go [hunk 2]
```diff
@@ -55,19 +55,19 @@ var channelRegExp = regexp.MustCompile(`^\/?channels\/([\w\-]+)\/messages(\/[^?]
 
 // Event implements events.Event interface.
 type handler struct {
-	publishers []messaging.Publisher
```
Comment (SammyOina) [inline mqtt/handler.go:58]: why remove support for mutliple publishers

File: pkg/messaging/mqtt/pubsub.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ import (
 
 const (
 	username = "mainflux-mqtt"
-	qos      = 2
+	qos      = 1 // At least once - Since NATS supports up to QoS 1.
```
Comment (SammyOina) [inline pkg/messaging/mqtt/pubsub.go]: should be configurable with each profile or via env or some other way to pass as an argument

**Review by SammyOina — CHANGES_REQUESTED** [review]

File: mqtt/handler.go (hunk 2 above)
Comment (rodneyosodo) [reply mqtt/handler.go:58]: It was not being utilized. We only used passed one publisher.

File: pkg/messaging/rabbitmq/pubsub_test.go (hunk 1 above)
Comment (rodneyosodo) [reply pkg/messaging/rabbitmq/pubsub_test.go:422]: The published message only contains `channel` and `payload`

**Review by dborovcanin — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM
